datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1058,6 +1065,65 @@ pub async fn from_substrait_rel(
                 partitioning_scheme,
             }))
         }
+        Some(RelType::Write(write)) => {
+            let table_name = match &write.write_type {
+                Some(WriteType::NamedTable(now)) => from_substrait_names(&now.names)?,
+                Some(WriteType::ExtensionTable(_)) => {
+                    return not_impl_err!("Unsupported WriteType: ExtensionTable");
+                }
+                _ => {
+                    return plan_err!("No WriteType specified in WriteRel");
+                }
+            };
+
+            let table_schema = from_substrait_named_struct(
+                &write.table_schema.clone().unwrap(),
+                extensions,
+            )?;
+
+            let op = match write.op() {
+                write_rel::WriteOp::Insert => dml::WriteOp::Insert(dml::InsertOp::Append),
+                _ => {
+                    return not_impl_err!("Unsupported WriteOp: {:?}", write.op());
+                }
+            };
```

Comment:
minor/nit: if we move this up above `table_schema` we can avoid doing redundant work in the event that his validation fails.